tensorboard/webapp/widgets/line_chart_v2/lib/integration_test.ts

```diff
@@ -0,0 +1,197 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {MainThreadChart, TEST_ONLY} from './chart';
+import {DataSeriesMetadata} from './internal_types';
+import {
+  ChartCallbacks,
+  DataSeries,
+  RendererType,
+  ScaleType,
+} from './public_types';
+
+function createSeries(
+  id: string,
+  pointFn: (index: number) => number = Math.sin
+): DataSeries {
+  return {
+    id,
+    points: new Array(10).fill({x: 0, y: 0}).map((data, index) => {
+      return {x: index, y: pointFn(index)};
+    }),
+  };
+}
+
+function buildMetadata(metadata: Partial<DataSeriesMetadata>) {
+  return {
+    id: 'line',
+    displayName: 'Line',
+    visible: true,
+    color: '#000',
+    ...metadata,
+  };
+}
+
+describe('line_chart_v2/lib/integration test', () => {
+  let dom: SVGElement;
+  let callbacks: ChartCallbacks;
+  let chart: MainThreadChart;
+  let rafSpy: jasmine.Spy;
+
+  beforeEach(() => {
+    rafSpy = spyOn(TEST_ONLY.util, 'requestAnimationFrame').and.callFake(
+      (cb: FrameRequestCallback) => {
+        cb(0);
+        return 0;
+      }
+    );
+    dom = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    callbacks = {
+      onDrawEnd: jasmine.createSpy(),
+    };
+    chart = new MainThreadChart({
+      type: RendererType.SVG,
+      container: dom,
+      callbacks,
+      domDimension: {width: 200, height: 100},
+    });
+    chart.setXScaleType(ScaleType.LINEAR);
+    chart.setYScaleType(ScaleType.LINEAR);
+  });
+
+  describe('render', () => {
+    it('renders data', () => {
+      chart.updateMetadata({
+        line: buildMetadata({id: 'line', visible: true}),
+      });
+      chart.updateViewBox({x: [0, 10], y: [0, 10]});
+
+      expect(dom.children.length).toBe(0);
+
+      chart.updateData([createSeries('line', (index) => index)]);
+
+      expect(dom.children.length).toBe(1);
+      const path = dom.children[0] as SVGPathElement;
+      expect(path.tagName).toBe('path');
+      expect(path.getAttribute('d')).toBe(
+        'M0,100L20,90L40,80L60,70L80,60L100,50L120,40L140,30L160,20L180,10'
```

Comment:
Entirely optional, but if you like, you could do something like

```javascript
'M0,100 L20,90 L40,80 L60,70 L80,60 L100,50 L120,40 L140,30 L160,20 L180,10'.replaceAll(' ', '')
```

for readability.